client/blocks/domain-to-plan-nudge/index.jsx

```diff
@@ -0,0 +1,333 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import React, { PropTypes, Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import DismissibleCard from 'blocks/dismissible-card';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSite, isCurrentSitePlan } from 'state/sites/selectors';
+import { plansList, PLAN_FREE, PLAN_PERSONAL } from 'lib/plans/constants';
+import { storedCardPayment } from 'lib/store-transactions';
+import { getStoredCards } from 'state/stored-cards/selectors';
+import QueryStoredCards from 'components/data/query-stored-cards';
+import { emptyCart } from 'lib/cart-values';
+import { add } from 'lib/cart-values/cart-items';
+import { submitTransaction } from 'lib/upgrades/actions/checkout';
+import PlanPrice from 'my-sites/plan-price';
+import PlanIcon from 'components/plans/plan-icon';
+import Gridicon from 'components/gridicon';
+import { errorNotice, infoNotice, removeNotice } from 'state/notices/actions';
+import { getCurrentUserCurrencyCode } from 'state/current-user/selectors';
+import { recordOrder } from 'lib/analytics/ad-tracking';
+import { getPlanRawPrice } from 'state/plans/selectors';
+import {
+	getPlanDiscountedRawPrice,
+	getPlanRawDiscount,
+	getPlansBySiteId
+} from 'state/sites/plans/selectors';
+import QuerySitePlans from 'components/data/query-site-plans';
+import formatCurrency from 'lib/format-currency';
+
+const debug = debugFactory( 'calypso:domain-to-plan-nudge' );
+
+class DomainToPlanNudge extends Component {
+
+	constructor() {
+		super( ...arguments );
+		this.oneClickUpgrade = this.oneClickUpgrade.bind( this );
+		this.handleTransactionComplete = this.handleTransactionComplete.bind( this );
+		this.state = {
+			isSubmitting: false,
+			noticeId: null
+		};
+	}
+
+	static propTypes = {
+		hasFreePlan: PropTypes.bool,
+		site: PropTypes.object,
+		siteId: PropTypes.number,
+		translate: PropTypes.func
+	};
+
+	isVisible() {
```

Comment:
Good catch!

I like your suggestions. To add some more: `shouldRenderNudge` (or just `shouldRender` as we are in the nudge's codebase so 'nudge' is implicit).

<bikeshed />